packages/vitest/src/runtime/mocker.ts

```diff
@@ -360,10 +360,12 @@ export class VitestMocker {
       const cache = this.moduleCache.get(mockPath)
       if (cache?.exports)
         return cache.exports
-      const cacheKey = toFilePath(dep, this.root)
-      const mod = this.moduleCache.get(cacheKey)?.exports || await this.request(dep)
-      const exports = this.mockObject(mod)
+
+      const exports = {}
+      // Assign the empty exports object early to allow for cycles to work. The object will be filled by mockObject()
       this.moduleCache.set(mockPath, { exports })
+      const mod = await this.request(dep)
+      this.mockObject(mod, exports)
```

Comment:
I opted to provide the exports object to `mockObject()` to prevent having to copy the object once more here.

So this could also be 
```js
Object.assign(exports, this.mockObject(mod))
```